jscomp/super_errors/super_typecore.ml

```diff
@@ -228,8 +228,28 @@ let report_error env ppf = function
             type_expr typ;
           fprintf ppf "@ @[It only accepts %i %s; here, it's called with more.@]@]"
                       acceptsCount (if acceptsCount == 1 then "argument" else "arguments")
-      | Tconstr ((Path.Pdot (((Pdot (Path.Pident {name="Js"}, "Internal", _))| (Pident {name="Js_internal"})), ("fn" | "meth"), _)), _, _)
-        -> fprintf ppf "This is an uncurried bucklescript function. It must be applied with [@bs]."
+      | Tconstr (
+          (Path.Pdot (((Pdot (Path.Pident {name="Js"}, "Internal", _)) | (Pident {name="Js_internal"})), ("fn" | "meth"), _)), 
+          _, 
+          _
+        )
+        -> 
+          if !Super_reason_flag.using_reason_syntax then 
```

Comment:
I suppose this module only print reason syntax, lets keep things simple, make improved error message only work for reason syntax ? 
Currently, we did not activate better error message for ocaml syntax. In that case we don't need add a new state variable here

Reply:
The reason syntax printer is in a different module. This module’s logic is shared with typecore. Also, I’m pretty positive lots of pure ocaml projects enable super errors manually. Bsb is also in vscode and atom plugins’ configurations now, so we can’t skim on these features imo

Reply:
Can we do it in a separate PR so that we can merge this strictly better PR quickly -- I think we already had something relevant to whether reason syntax is enabled, but not sure where

Reply:
How do I split the PR? Also, the only thing I've seen that knows whether it's a reason file is `Reason_outcome_printer_main.setup()`. I can't put the flag there because of cyclic dependencies.

Reply:
I mean just always print it in reason syntax (not introducing a ref variable), does it work?

Reply:
We can't, because pure ocaml bs projects are using super-errors right now

Reply:
The reason setup logic is already pretty complex (to be honest, I don't 100% understand how they interact).

My impression is that super-errors is currently only activated for reason syntax, let's not over-promise, having a cursory look at super_typecore module, `print_arguments` is reason specific, I am not really sure this module is not reason syntax specific.

I think we should not make the setup logic more and more complex to deliver things we did not promise

Reply:
We're not over-promising, we're just not breaking existing experience. That print_arguments has one reason syntax-specific line because I missed it, but it's easy to make it work with ocaml syntax too.

Take a look at this result: https://github.com/search?q=-bs-super-errors&type=Code&utf8=✓
Some prominent projects, including the graphql integration (pure ocaml), uses super-errors.

Reply:
I don't think we're over-promising here. We've never really had reason-specific logic in super-errors (printing the outcome tree in reason is independent of super-errors. It's used elsewhere too), except that print_arguments which is my fault